apiml-common/src/test/java/org/zowe/apiml/product/eureka/client/ReplicationTaskProcessorTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.product.eureka.client;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.eureka.registry.PeerAwareInstanceRegistryImpl.Action;
+import com.netflix.eureka.util.batcher.TaskProcessor.ProcessingResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.net.ssl.SSLException;
+import java.util.Collections;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.zowe.apiml.product.eureka.client.ApimlPeerEurekaNode.ReplicationTaskProcessor;
+import static org.zowe.apiml.product.eureka.client.TestableInstanceReplicationTask.ProcessingState;
+import static org.zowe.apiml.product.eureka.client.TestableInstanceReplicationTask.aReplicationTask;
+
+public class ReplicationTaskProcessorTest {
+
+    private final TestableHttpReplicationClient replicationClient = new TestableHttpReplicationClient();
+
+    private ReplicationTaskProcessor replicationTaskProcessor;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        replicationTaskProcessor = new ReplicationTaskProcessor("peerId#test", replicationClient);
+    }
+
+    @Test
+    public void testNonBatchableTaskExecution() throws Exception {
```

Comment:
Could you use `@Nested` for this test and ApimlPeerEurekaNodeTest.java?

Reply:
added nested classes